TESTING.md

```diff
@@ -169,8 +169,102 @@ cargo make \
   test
 ```
 
-(`-r` tells testsys to also show the status of resources like the cluster and instances in addition to tests):
-
 ```shell
 cargo make watch-test
 ```
+
+## Migration Testing
+
+Migration testing starts testsys at a starting version, or a provided initial image id and migrates instances to the target version.
+In order to accomplish this a few artifacts need to be created:
+* `TUF` repos that are publicly accessible
+* A previous release of bottlerocket signed with available keys
+* The ami id from the previous release
+* Build images and repos for current changes
+
+### The setup
+
+#### Prepare `Infra.toml`
+
+A publicly available s3 bucket is needed to complete the next steps, once a bucket is created the steps may be continued.
+`Infra.toml` is used by testsys to determine tuf repo locations, so `metadata_base_url` and `targets_base_url` need to be set based on the s3 bucket. The examples below assume `metadata_base_url` is the top level directory of the bucket and `targets_base_url` is a directory named `targets` in the top level directory. 
+The examples below also assume that the default repo in `Infra.toml`, but any repo can be used by setting the `PUBLISH_REPO` environment variable.
+
+#### Starting bottlerocket images
+
+In this example we will use `v1.8.0` as our starting bottlerocket version, but any tag from bottlerocket will work. 
+The following bash will checkout the proper branch from git and create the build images and repos for testing.
+
+```shell
+export TESTSYS_STARTING_VERSION="1.8.0"
+git checkout v${TESTSYS_STARTING_VERSION}
+cargo make
+cargo make ami
+cargo make repo
+```
+
+Next, the repos need to be added to the `TUF` repos (s3 bucket). Make sure to change `S3_BUCKET_NAME` to the `TUF` repo bucket name.
+
+```shell
+S3_BUCKET_NAME=<s3 bucket name>
+aws s3 sync build/repos/default/latest s3://${S3_BUCKET_NAME}/
+```
+
+The last step with the older bottlerocket version is retrieving the ami that was created.
+
+```shell
+BUILDSYS_ARCH="x86_64"
+BUILDSYS_VARIANT="aws-k8s-1.21"
+REGION="us-west-2"
+
+AMIS_JSON="build/images/${BUILDSYS_ARCH}-${BUILDSYS_VARIANT}/latest/bottlerocket-${BUILDSYS_VARIANT}-${BUILDSYS_ARCH}-amis.json"
+export TESTSYS_STARTING_IMAGE_ID=$(jq -r --arg r "${REGION}" ' .[$r].id' "${AMIS_JSON}")
+echo "Starting ami for migration tests '${TESTSYS_STARTING_IMAGE_ID}'"
+```
+
+Make note of the ami that is printed, it will be needed in future steps.
+
+#### Target bottlerocket images
+
+Now, it's time to create the bottlerocket artifacts that need to be upgraded to.
+
+Switch to the working git branch that should be built from.
+
+```shell
+WORKING_BRANCH="develop"
+git checkout ${WORKING_BRANCH}
+```
+
+Now, [Release.toml](Release.toml) may need to be updated.
```

Comment:
For this testing it must, right?

Reply:
For using develop yes. But if someone wanted to try building 1.6.0 and 1.6.1 to see how testsys migrations work they wouldn't need to.